### PR TITLE
[Fix] `order`: ensure arcane imports do not cause undefined behavior

### PR DESCRIPTION
**Pull request by Xunnamius**
Depends on #3127

**This PR implements in `import/order`: a fix to ensure `NaN` is never passed into rank-computing functions, which can result in undefined behavior.**

A demo package containing this fix is temporarily available for easy testing:

```bash
npm install --save-dev eslint-plugin-import@npm:@-xun/eslint-plugin-import-experimental
```

## Ensure strange imports do not cause strange behavior

There are certain edge cases where a `NaN` rank gets passed around, such as using an import with an absolute specifier under certain configurations. The fix concerns the `computeRank` function.

This PR includes a unit test to catch potential regressions.

### CODE REVIEW

File: src/rules/order.js [hunk 1]
```diff
@@ -526,8 +526,13 @@ function computeRank(context, ranks, importEntry, excludedImportTypes) {
   if (!excludedImportTypes.has(impType)) {
     rank = computePathRank(ranks.groups, ranks.pathGroups, importEntry.value, ranks.maxPosition);
   }
-  if (typeof rank === 'undefined') {
+
+  if (rank === undefined) {
     rank = ranks.groups[impType];
+
+    if(rank === undefined) {
```
Comment (ljharb) [inline src/rules/order.js:533]:
```suggestion
  if (typeof rank === 'undefined') {
    rank = ranks.groups[impType];

    if (typeof rank === 'undefined') {
```

**Review by ljharb — COMMENTED** [review]
seems good overall, assuming the test fails without the fix :-)

File: src/rules/order.js (hunk 1 above)
Comment (Xunnamius) [reply src/rules/order.js]: I hoped I could factor this change out from contrib-sort-types-among-themselves, but I've noticed the test needs `sortTypesGroup` to trigger the bad behavior :)

**Comment by Xunnamius** [comment]
It's a little interesting that codecov believes this PR does not reduce coverage, this PR that includes the changes from the two preceding PRs... both of which codecov claims have reduced coverage 😄

**Review by ljharb — APPROVED** [review]
